### PR TITLE
Feature add events

### PR DESCRIPTION
**Pull request by fezzopro**
## PR Details

This pull request contains the logic and event handler for the calculator react app.

More information can be found **[HERE](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_events.md)**

## Screenshot

image

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+          {sign}
+        </button>
+      );
+    });
+  };
+  const displayValue = `${state.total ?? ''} ${state.operation ?? ''} ${state.next ?? ''}`;
+
   const signs = ['AC', '+/-', '%', '+', '7', '8', '9', '*', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
   return (
     <div className="calculator">
```
Comment (DuaneDave) [inline src/components/Calculator.js:30]: The operations are expected to work seamlessly, but currently when I try to multiply, the app would not work, instead it breaks and also, I noticed that you do not have the division button, instead the addition is duplicated. Consider fixing the above mentioned issue 

cal

fd

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @fezzopro  👋 , 

Good job so far on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ You displayed a good understanding of states and the methodology of updating them


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js [hunk 2]
```diff
@@ -1,7 +1,30 @@
...
+import { useState } from 'react';
+import calculate from '../logic/calculate';
 
 const Calculator = () => {
-  const displayValue = 0;
+  const [state, setState] = useState({ next: null, operation: null, total: null });
+  const Button = (props) => {
+    const signs = props.signs; //eslint-disable-line
+    return signs.map((sign) => {  //eslint-disable-line
+      const callCalculate = (value) => {
```
[the remark this answers is not in the document]
Comment (fezzopro) [reply src/components/Calculator.js]: The comments have been removed 🙏

File: src/components/Calculator.js (hunk 1 above)
Comment (fezzopro) [reply src/components/Calculator.js]: This * operator has been changed to `x` operator, and one `+` operator was replaced by `÷` operator.
Thanks for spotting this error 🙏

**Pull request merged**